scripts/one-offs/DR-2796-backfill-more-dates/backfill_additional_dates.rb

```diff
@@ -0,0 +1,173 @@
+# frozen_string_literal: false
+
+# This comes from https://jira.nypl.org/browse/DR-2796
+
+# We've updated a lot of things to have string dates, and we've added a lot of database records for first indexed dates. This should make sure we have actual dates for all first indexed values. 
+
+# How to run: ruby ./scripts/one-offs/DR-2796-backfill-more-dates/backfill_additional_dates.rb YOUR_FAVORITE_SOLR_URL
+# e.g. `ruby ./scripts/one-offs/DR-2796-backfill-more-dates/backfill_additional_dates.rb http://10.225.133.217:8983/solr/repoapi`
+# It will run for a long time and might need to be restarted a bunch, but as it will work through all solr docs missing dates, that runtime should decrease over time. It is not a problem to start and restart this script. 
+
+require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', '..', 'config', 'environment'))
+require 'time'
+require 'csv'
+
+def check_missing_indexed_capture_docs(solr_url)
+  solr = RSolr.connect(url: solr_url)
+  rows_per_page = 250 # Adjust as needed based on your Solr setup
+  
+  start = 0
+  total_results = nil
+  missing_uuids = []
+
+  loop do
+    # Construct Solr query to find documents missing firstIndexed_dt with a title (prevents looking at deactivated records)
+    # We also need to loop first through captures as they are the ones that will have values associated with dates from filestore. 
+    query = '-firstIndexed_dt:[* TO *] AND mainTitle:[ * TO * ] AND type_s:Capture'
+    
+    # Perform Solr search with pagination
+    response = solr.get('select', params: {
+      q: query,
+      fl: 'uuid',
+      rows: rows_per_page,
+      start: start
+    })
+
+    total_results ||= response['response']['numFound']
+    
+    update_json = []
+
+    # Loop through each document
+    response['response']['docs'].each do |doc|
+      uuid = doc['uuid']
+
+      # Check if corresponding record exists in RepoSolrDoc
+      unless RepoSolrDoc.exists?(uuid: uuid)
```

Comment:
My local ingest DB doesn't have any of these.

 Is it possible that any records were pushed to solr straight from mms and wouldn't be in the production ingest database, or did we do the last full index via ingest?

Reply:
Is this script meant to run on the production server or from someone's local?

Reply:
This script is flexible, could be run from local, but I was thinking server. 

I can't say I expect any records to be missing from production database (and haven't found any yet), but I was being cautious?